lib/rules/validate-quote-marks.js

```diff
@@ -106,10 +106,16 @@ module.exports.prototype = {
                     return;
                 }
 
+                token.value = token.value.replace(
+                    new RegExp('^' + mark + '(.*)' + mark + '$'),
```

Comment:
I'd would use something time-tested, like https://github.com/sindresorhus/to-single-quotes/blob/master/index.js